### PR TITLE
Cake 4.x Stable

### LINKED ISSUE #290: CakePHP 4.x Compatibility
I'm looking to upgrade an application to CakePHP 4.0, but we rely on CakePDF to generate PDFs. 

Composer is throwing the obvious errors that there isn't a compatible release and I noticed the 4.x branch hasn't been touched in a while.

Just wondering if there's anything I can do or if a quick release could be made. 

Thanks for all the hard work! :)

### PR DESCRIPTION
**Pull request by kyleweishaupt**
Update CakePHP dependency to 4.0 stable. Closes #290

### CODE REVIEW

File: tests/TestCase/Pdf/CakePdfTest.php [hunk 1]
```diff
@@ -97,7 +97,7 @@ public function testCustomTemplatePaths($config)
         $pdf->template('view', 'default');
         $pdf->templatePath('Posts/pdf');
         $result = $pdf->output();
-        $expected = '<h2>Rendered with default layout</h2>' . "\n" . 'Post data: testing';
+        $expected = '<h2>Rendered with default layout</h2>' . "\r\n" . 'Post data: testing';
         $this->assertEquals($expected, $result);
```
Comment (dereuromark) [inline tests/TestCase/Pdf/CakePdfTest.php]: You can also use assertTextEquals() and dont have to change the line endings - will be cross OS safe.

**Comment by dereuromark** [comment]
FILE: /home/travis/build/FriendsOfCake/CakePdf/src/Pdf/Engine/MpdfEngine.php
--------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------------
 20 | ERROR | [x] The first expression of a multi-line control structure must
    |       |     be on the line after the opening parenthesis

one cs error

  Line   Pdf/CakePdf.php                                                 
 ------ ---------------------------------------------------------------- 
  958    Static method Cake\Routing\Router::getRequest() invoked with 1  
         parameter, 0 required.                                          
 ------ --------

one phpstan error and all is green.

**Comment by dereuromark** [comment]
We will also need to merge master into 4.x later
https://github.com/FriendsOfCake/CakePdf/commit/05aa0bfc2a9eaa58e899be94385718079833d97b
then it should be releasable as beta.

**Comment by ADmad** [comment]
If have taken care of fixing errors reported by static analyzers and adding typhints where possible after this merge.

The `CakePdf` class using many combined getter/setter methods. Is anyone interested in splitting them into separate methods?  If not I can do a beta release.

**Pull request merged**